app/services/profiles/extract_data.rb

```diff
@@ -1,45 +1,12 @@
 module Profiles
   module ExtractData
-    DIRECT_ATTRIBUTES = %i[
-      available_for
-      behance_url
-      currently_hacking_on
-      currently_learning
-      dribble_url
-      education
-      employer_name
-      employer_url
-      employment_title
-      facebook_url
-      instagram_url
-      location
-      looking_for_work
-      mastodon_url
-      medium_url
-      mostly_work_with
-      name
-      summary
-      twitch_url
-      website_url
-      youtube_url
-    ].freeze
-
-    MAPPED_ATTRIBUTES = {
-      brand_color1: :bg_color_hex,
-      brand_color2: :text_color_hex,
-      display_email_on_profile: :email_public,
-      display_looking_for_work_on_profile: :looking_for_work_publicly,
-      git_lab_url: :gitlab_url,
-      linked_in_url: :linkedin_url,
-      recruiters_can_contact_me_about_job_opportunities: :contact_consent,
-      stack_overflow_url: :stackoverflow_url
-    }.freeze
-
     def self.call(user)
-      user_attributes = user.attributes.symbolize_keys!
+      user_attributes = user.attributes
 
-      direct_data = user_attributes.extract!(*DIRECT_ATTRIBUTES)
-      mapped_data = MAPPED_ATTRIBUTES.keys.zip(user_attributes.values_at(*MAPPED_ATTRIBUTES.values)).to_h
+      mapped_attributes = Profile::MAPPED_ATTRIBUTES
+      direct_attributes = Profile.attributes! - mapped_attributes.keys
+      direct_data = user_attributes.extract!(*direct_attributes)
+      mapped_data = mapped_attributes.keys.zip(user_attributes.values_at(*mapped_attributes.values)).to_h
```

Comment:
The `mapped_attributes.values` here need to be converted to strings otherwise you wont be able to find them in user_attributes which results in nothing getting set in the Profile. 
```ruby
mapped_data = mapped_attributes.keys.zip(user_attributes.values_at(*mapped_attributes.values.map(&:to_s))).to_h
```

I figured this out by pulling the branch down and testing it and seeing that fields were not getting updated in the Profile from the user. I think we should write at least 1 end-to-end test where we check if a mapped user value is updated, after saving the profile is also updated.

Reply:
It seems to be working fine for me:

[6] forem(main)>
[6] forem(main)> user.name
"Hi Molly 👋"
[7] forem(main)> user.profile.name
"Hi Molly 👋"
[8] forem(main)> user.update(name: "New name, who dis?")
...
[9] forem(main)> user.name
"New name, who dis?"
[10] forem(main)> user.profile.name
"New name, who dis?"
```

There also is a test for this:

```ruby
    it "propagates changes to the profile model", :aggregate_failures do
      expect do
        user.update(available_for: "profile migrations")
      end.to change { user.profile.reload.available_for }.from(nil).to("profile migrations")

      # Changes were also persisted in the users table
      expect(user.reload.available_for).to eq "profile migrations"
    end
```

Reply:
Thanks @mstruve! Fixed the code and added a test for a mapped attribute.